src/librustdoc/html/render/search_index.rs

```diff
@@ -254,14 +253,12 @@ crate fn get_real_types<'tcx>(
     tcx: TyCtxt<'tcx>,
     recurse: usize,
     res: &mut Vec<TypeWithKind>,
-    cache: &Cache,
 ) {
     fn insert_ty(
         res: &mut Vec<TypeWithKind>,
         tcx: TyCtxt<'_>,
         ty: Type,
         mut generics: Vec<TypeWithKind>,
-        _cache: &Cache,
```

Comment:
I don't think this is the right move because of:

```rust
} else if let Some(kind) = ty.def_id_no_primitives().map(|did| tcx.def_kind(did).into()) {
```

We are replacing `def_id_no_primitives` calls, so we need to keep the `Cache`.

Reply:
Hmm, I see what you mean, but the `Cache` can easily be re-added later. As of now, passing around the `Cache` is just dead code that might be sitting there for a while.

Reply:
I reviewed a PR doing exactly that today (but of course, can't find it again...). I'd prefer to avoid having removals/adds following but not sure it matters much since it's just about git history here...

Reply:
I think you're thinking of #92342? I'd rather remove the dead code for now – e.g., that PR may not work and end up being closed, and then the dead code will still be there.

Reply:
I don't fully agree with you but it's fine. Let's move forward then!